nestkernel/node_collection.h

```diff
@@ -218,6 +218,19 @@ class NodeCollection
    */
   static NodeCollectionPTR create( const std::vector< index >& node_ids );
 
+  /**
+   * @brief Create a NodeCollection from an array of node IDs without requiring
+   * instantiation of a node model and associated managers. Intended to simplify
+   * C++ test programs where just a NodeCollection is needed. Since the fingerprint
+   * of the NodeCollection superclass require a kernel instance, a kernel must be
+   * instantiated, i.e., KernelManager::create_kernel_manager(), before this
+   * creator can be used.
+   *
+   * @param node_ids Array of node IDs from which to create the NodeCollection
+   * @return a NodeCollection pointer to the created NodeCollection
+   */
+  static NodeCollectionPTR create_test_collection( const std::vector< index >& node_ids );
```

Comment:
```suggestion
  static NodeCollectionPTR create_mock_node_collection( const std::vector< index >& node_ids );
```
Mocking is the technical term for incomplete instances like this.